video/clip.go

```diff
@@ -86,3 +98,24 @@ func ClipManifest(requestID string, manifest *m3u8.MediaPlaylist, startTime, end
 
 	return relevantSegments, nil
 }
+
+// Clips a segment by re-encoding so that I-frames are placed at the right intervals
+func ClipSegment(tsInputFile, tsOutputFile string, startTime, endTime float64) error {
+	start := formatTime(startTime)
+	end := formatTime(endTime)
+	err := ffmpeg.Input(tsInputFile).
+		Output(tsOutputFile, ffmpeg.KwArgs{"ss": start, "to": end,
+			"c:v":          "libx264",
+			"r":            "24",
+			"g":            "48",
+			"keyint_min":   "48",
+			"sc_threshold": 50,
+			"bf":           "0",
+			"c:a":          "copy"}).
+		OverWriteOutput().ErrorToStdOut().Run()
```

Comment:
Note: these settings will likely be revisited to be optimized before we release clipping. Using these reasonable defaults for now to finish integration.